chall01/srouhe.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/python3
+
+import sys
+
+MAPPING = {
+	'a': '.-',
+	'b': '-...',
+	'c': '-.-.',
+	'd': '-..',
+	'e': '.',
+	'f': '..-.', 
+	'g': '--.', 
+	'h': '....', 
+	'i': '..', 
+	'j': '.---', 
+	'k': '-.-', 
+	'l': '.-..', 
+	'm': '--', 
+	'n': '-.', 
+	'o': '---', 
+	'p': '.--.', 
+	'q': '--.-', 
+	'r': '.-.', 
+	's': '...', 
+	't': '-', 
+	'u': '..-', 
+	'v': '...-', 
+	'w': '.--', 
+	'x': '-..-', 
+	'y': '-.--', 
+	'z': '--..',
+	' ': ' '
+}
+
+def print_usage():
+	print('usage: ./srouhe.py <a-zA-Z string>')
+	sys.exit(1)
+
+def encode_arg(s):
+	r = ''
+	for c in s:
+		try:
+			r += MAPPING[c]
+		except KeyError:
+			print_usage()
```

Comment:
you could avoid try / except for this simple script by making sure you handle all the error case possible (there is not so many)